cmd/appliance/upgrade/complete.go

```diff
@@ -281,16 +281,23 @@ func upgradeCompleteRun(cmd *cobra.Command, args []string, opts *upgradeComplete
 	}
 
 	// isolate log forwarders and log servers
-	logForwardersAndServers := groups[appliancepkg.FunctionLogServer]
-	for _, lf := range groups[appliancepkg.FunctionLogForwarder] {
-		logForwardersAndServers = appliancepkg.AppendUniqueAppliance(logForwardersAndServers, lf)
-	}
-	for _, lfs := range logForwardersAndServers {
+	logForwardersAndServersAll := append(groups[appliancepkg.FunctionLogServer], groups[appliancepkg.FunctionLogForwarder]...)
+	logForwardersAndServers := []openapi.Appliance{}
+	for _, lfs := range logForwardersAndServersAll {
 		for i, app := range additionalAppliances {
 			if lfs.GetId() == app.GetId() {
 				additionalAppliances = append(additionalAppliances[:i], additionalAppliances[i+1:]...)
 			}
 		}
+		isAlsoInControllers := false
+		for _, ctrl := range additionalControllers {
+			if lfs.GetId() == ctrl.GetId() {
+				isAlsoInControllers = true
+			}
+		}
```

Comment:
Whenever I see this pattern, I wish Go provided a .contains() method for slices in their standard library :P